quality_check/08-gene_lengths.Rmd

```diff
@@ -0,0 +1,293 @@
+rnaseq_df <- 
+  rnaseq_df[, c(TRUE, !grepl("Gene", colnames(rnaseq_df))[2:ncol(rnaseq_df)])]
+rnaseq_df <- rnaseq_df %>%
+  as.data.frame() %>%
+  tibble::column_to_rownames("Gene")
+# remove the list of individual sample data.frame
+rm(rnaseq_df_list)
+```
+
+```{r}
+# read in gene lengths
+gene_lengths_df <- 
+  readr::read_tsv(file.path("data", 
+                            "Danio_rerio.GRCz11_gene_lengths_exons.tsv"))
+```
+
+## Gene zero counts and expression levels
+
+### For each gene, how many samples have a count-scale expression value of zero?
+
+```{r}
+# calculate the proportion of samples a gene is zero for
+gene_zeroes_proportion <- data.frame(
+  "Gene" = rownames(rnaseq_df),
+  "Zero_proportion" = apply(rnaseq_df, 1, 
+                            function(x) sum(x == 0) / length(x))
+)
+```
+
+```{r}
+# bind to lengths, this will be used for plotting the relationship between
+# gene length and number of zeroes
+zeroes_df <- dplyr::inner_join(gene_lengths_df, gene_zeroes_proportion,
+                               by = "Gene")
+```
+
+```{r}
+# there are some really long genes that will probably make it difficult to
+# visualize what is going on
+summary(zeroes_df$gene_length)
+```
+
+#### All genes
+
+```{r}
+# filtering genes based on length to get a clearer picture
+zeroes_df %>%
+  dplyr::filter(gene_length <= 20000) %>%
+  ggplot(aes(x = gene_length, y = Zero_proportion)) +
+  geom_point(alpha = 0.2) +
+  geom_smooth() +
+  theme_bw() +
+  labs(x = "gene length", y = "proportion of samples with zero count",
+       subtitle = "All genes", title = "Gene length and observations")
+```
+
+```{r}
+ggsave(filename = file.path(plot_dir, 
+                            "gene_length_vs_zero_count_all_genes.png"),
+       plot = last_plot())
+```
+
+#### Compendium genes only
+
+```{r}
+# subset just to genes that are in the compendium
+zeroes_df %>%
+  dplyr::filter(Gene %in% compendium_genes, 
+                gene_length <= 20000) %>%
+  ggplot(aes(x = gene_length, y = Zero_proportion)) +
+  geom_point(alpha = 0.2) +
+  geom_smooth() +
+  theme_bw() +
+  labs(x = "gene length", y = "proportion of samples with zero count",
+       subtitle = "Test compendium genes only", 
+       title = "Gene length and observations")
+```
+
+```{r}
+ggsave(filename = file.path(plot_dir, 
+                            "gene_length_vs_zero_count_compendium_genes.png"),
+       plot = last_plot())
+```
+
+#### Takeaway
+
+Shorter genes do tend to have a larger proportion of samples where they have
+zero values. 
+It stands to reason that if we do not _observe_ shorter genes, we can not
+correct for length-bias effectively. 
+(Recall that `lengthScaledTPM` are calculated using the average tx length
+observed across samples.)
+
+## Removing "short" genes from the test species compendium
+
+Does removing shorter genes from the species compendium remove evidence of a
+technology bias?
+
+```{r}
+exprs_mat <- compendium_exprs_df %>%
```

Comment:
This looks like stuff you had from before, so just apply whatever you changes you did previously to here.

Reply:
I'll move to loading the RDS like I did in #19

Reply:
Whoops, no that's a different compendium. This will get tweaked though.